src/lazy/encoder/binary/v1_1/flex_uint.rs

```diff
@@ -46,79 +45,49 @@ impl FlexUInt {
+        Self::read_flex_primitive_as_uint(input, offset, "reading a FlexUInt")
     }
 
-    /// Helper method that reads a [`FlexUInt`] with 7 or fewer bytes of magnitude from the buffer.
-    // Caller must confirm that `bytes` has at least 8 bytes.
     #[inline]
-    fn read_small_flex_uint(bytes: &[u8]) -> FlexUInt {
-        debug_assert!(bytes.len() >= 8);
-        let num_encoded_bytes = bytes[0].trailing_zeros() as usize + 1;
-        let num_encoded_bits = 8 * num_encoded_bytes;
-        // Get a mask with the low 'n' bits set
-        // TODO: Should this be a const cache of num_encoded_bits -> mask?
-        let mask = 1u64
-            .checked_shl(num_encoded_bits as u32)
-            .map(|v| v - 1)
-            .unwrap_or(u64::MAX);
-        // Convert our longer-than-8-bytes slice to a fixed sized 8-byte array that we can convert
-        // to a u64 directly.
-        let fixed_size_input: [u8; 8] = bytes[..8].try_into().unwrap();
-        // This step will often read unrelated bytes from beyond the FlexUInt, but they are
-        // discarded in the shift operation that follows.
-        let encoded_value = u64::from_le_bytes(fixed_size_input);
-        // Note that `num_encoded_bytes` is also the number of continuation flags that we need
-        // to discard via right shifting.
-        let value = (encoded_value & mask) >> num_encoded_bytes;
-        FlexUInt::new(num_encoded_bytes, value)
-    }
-
-    #[inline(never)]
-    pub(crate) fn read_flex_primitive_as_uint_no_inline(
-        input: &[u8],
-        offset: usize,
-        label: &'static str,
-        support_sign_extension: bool,
-    ) -> IonResult<FlexUInt> {
-        Self::read_flex_primitive_as_uint(input, offset, label, support_sign_extension)
+    pub fn read_n_bytes<const NUM_BYTES: usize>(bytes: &[u8], buffer: &mut [u8; size_of::<u64>()]) {
+        let input: [u8; NUM_BYTES] = *(bytes.first_chunk::<NUM_BYTES>().unwrap());
+        *buffer.first_chunk_mut::<NUM_BYTES>().unwrap() = input;
     }
 
-    /// Helper method that reads a flex-encoded primitive from the buffer, returning it as a `FlexUInt`.
-    /// If an error occurs while reading, its description will include the supplied `label`.
-    ///
-    /// The current implementation supports flex primitives with up to 64 bits of representation
-    /// beyond the leading header bits. Flex primitives requiring 10 bytes to encode have 70 magnitude
-    /// bits. If this value is unsigned (`support_sign_extension=false`), the six bits beyond the
-    /// supported 64 must all be `0`. If this value will later be re-interpreted as a signed value,
-    /// (`support_sign_extension=true`), then the six bits beyond the supported 64 must all be the
-    /// same as the 64th (highest supported) bit. This will allow encodings of up to 70 bits
-    /// to be correctly interpreted as positive, negative, or beyond the bounds of the 64 bit
-    /// limitation.
     pub(crate) fn read_flex_primitive_as_uint(
```

Comment:
🪧 This was the general-purpose decoder, handling incomplete data and magnitudes greater than 7 bytes. Because the serialized form couldn't fit in a `u64` (and Rust didn't yet have a `u128`), we had to manipulate a byte array on the stack to achieve the desired effect.